irr/src/OpenGL/Driver.cpp

```diff
@@ -477,44 +477,36 @@ void COpenGL3DriverBase::setTransform(E_TRANSFORMATION_STATE state, const core::
 	Transformation3DChanged = true;
 }
 
-bool COpenGL3DriverBase::updateVertexHardwareBuffer(SHWBufferLink_opengl *HWBuffer)
+bool COpenGL3DriverBase::updateHardwareBuffer(SHWBufferLink_opengl *HWBuffer,
+	const void *buffer, size_t bufferSize, scene::E_HARDWARE_MAPPING hint)
 {
-	if (!HWBuffer)
-		return false;
+	assert(HWBuffer);
 
-	const scene::IMeshBuffer *mb = HWBuffer->MeshBuffer;
-	const void *vertices = mb->getVertices();
-	const u32 vertexCount = mb->getVertexCount();
-	const E_VERTEX_TYPE vType = mb->getVertexType();
-	const u32 vertexSize = getVertexPitchFromType(vType);
-
-	const void *buffer = vertices;
-	size_t bufferSize = vertexSize * vertexCount;
 	accountHWBufferUpload(bufferSize);
 
 	// get or create buffer
 	bool newBuffer = false;
-	if (!HWBuffer->vbo_verticesID) {
-		GL.GenBuffers(1, &HWBuffer->vbo_verticesID);
-		if (!HWBuffer->vbo_verticesID)
+	if (!HWBuffer->vbo_ID) {
+		GL.GenBuffers(1, &HWBuffer->vbo_ID);
+		if (!HWBuffer->vbo_ID)
 			return false;
 		newBuffer = true;
-	} else if (HWBuffer->vbo_verticesSize < bufferSize) {
+	} else if (HWBuffer->vbo_Size < bufferSize) {
 		newBuffer = true;
 	}
 
-	GL.BindBuffer(GL_ARRAY_BUFFER, HWBuffer->vbo_verticesID);
+	GL.BindBuffer(GL_ARRAY_BUFFER, HWBuffer->vbo_ID);
```

Comment:
As the "`vbo_ID`" can be an index buffer, maybe use `GL_COPY_WRITE_BUFFER`. (Doesn't make a difference, afaik. But it would be cleaner I think.)

Reply:
not available in gles 2.0
if we're gonna do something that requires fallback code using GL_EXT_direct_state_access would be more worth it as suggested [here](https://stackoverflow.com/questions/20391921/#answer-20407668)